improvement(settings): make fragments more independent

BrayanDSO opened the pull request:
## Purpose / Description

Key differences:
1. The fragments now contain their own toolbar and title, which fixed the delay of setting the title when changing the fragment
2. Big screens: each page has its tile above it instead of modifying the `headers page` title
3. Big screens: Reviewer Menu options now works as a fragment instead of an activity


## How Has This Been Tested?

Emulator API 35:

<details> <summary>First three commits</summary>

https://github.com/user-attachments/assets/ca3dbda6-cf42-4bfd-bb0f-68cf05114ea6

https://github.com/user-attachments/assets/b3f8c299-35ca-4be4-bfe4-ebc2448e4752

</details>

<details> <summary>Last two commits</summary>

https://github.com/user-attachments/assets/3c885cf5-b0ac-47ff-b1e1-5fd84515a89d

![Screenshot_20250703_114555](https://github.com/user-attachments/assets/49f532b2-9e74-4609-a480-c4a5147e7c4a)

</details>

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [X] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

<!--- Uncomment this section ONLY if this PR introduces new resources (external libraries, icons etc)
## Licenses
_For each new external resource, add a row to the table below:_

| Library | Description | License |
| --- | --- | --- |
| Sample Icon Library | Sample Description | [The Apache Software License, Version 2.0](http://www.apache.org/licenses/LICENSE-2.0.txt) |

**Maintainers:**

* [ ] Add the https://github.com/ankidroid/Anki-Android/labels/Licenses label
* [ ] Update the [licenses](https://github.com/ankidroid/Anki-Android/wiki/Licences) wiki when merging
--->

david-allison commented on AnkiDroid/src/main/java/com/ichi2/anki/preferences/Preferences.kt:
```diff
@@ -174,11 +161,55 @@ class PreferencesFragment :
-        view?.findViewById<CollapsingToolbarLayout>(R.id.collapsingToolbarLayout)?.title = title
-        view?.findViewById<MaterialToolbar>(R.id.toolbar)?.title = title
+        // Configure headers highlight
+        childFragmentManager.executePendingTransactions() // wait the headers page creation
```
wait for

david-allison reviewed (commented):
Tablet: New study screen - Study actions

Unsure on the back behaviour here. I'd expect 'Android back' to move me back to 'new study screen' and 'navigation click' to take me back to the deck picker

BrayanDSO commented:
> I'd expect 'Android back' to move me back to 'new study screen' and 'navigation click' to take me back to the deck picker

That was an existing problem with screens deeper in the navigation hierarchy, but probably more noticeable in `New study screen > Toolbar actions` because the user may want to change more things in the reviewer besides that.

Also added a divider. Updated the PR description.

david-allison reviewed (approved):
Even better, cheers!
